### PR TITLE
Trigger Service should support TLS Ledger Clients

### PR DESCRIPTION
**Pull request by carlpulley-da**
- [x] implement main changes to resolve https://github.com/digital-asset/daml/issues/17149

For a future PR:
- document how to configure trigger service (and probably the script service!) to leverage TLS with ledger clients

### CODE REVIEW

File: triggers/service/BUILD.bazel [hunk 1]
```diff
@@ -94,6 +94,7 @@ da_scala_library(
         "@maven//:ch_qos_logback_logback_classic",
         "@maven//:com_typesafe_config",
         "@maven//:io_dropwizard_metrics_metrics_core",
+        "@maven//:io_netty_netty_handler",
```
Comment (carlpulley-da) [inline triggers/service/BUILD.bazel:97]: Needed for access to `SslContext`

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerImpl.scala [hunk 2]
```diff
@@ -88,7 +90,7 @@ object TriggerRunnerImpl {
       )
 
       val channelConfig = LedgerClientChannelConfiguration(
-        sslContext = None,
+        sslContext = config.tlsConfig.client(),
```
Comment (carlpulley-da) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerImpl.scala:93]: Key change that allows trigger service ledger clients to use TLS

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceAppConf.scala [hunk 3]
```diff
@@ -80,6 +82,26 @@ private[trigger] object TriggerServiceAppConf {
...
+        certChainFile <- crtCur.asString
+        cacrtCur <- objCur.atKey("cacrt")
+        trustCollectionFile <- cacrtCur.asString
+      } yield TlsConfiguration(
+        enabled,
+        Some(new File(certChainFile)),
+        Some(new File(privateKeyFile)),
+        Some(new File(trustCollectionFile)),
+      )
+    }
```
Comment (carlpulley-da) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceAppConf.scala:103]: Allows TLS to be configured using Hocon blocks such as:
```
tls-config {
  tls = true
  pem = "some pem file path"
  crt = "some crt file path"
  cacrt = "some cacrt file path"
}
```

**Review by remyhaemmerle-da — COMMENTED** [review]
LGTM. Thanks

**Pull request merged**